samples/proxytool/proxytool-3.3.py

```diff
@@ -0,0 +1,197 @@
+                    required=False,
+                    default="auto"
+                    )
+
+args = parser.parse_args()
+
+if args.scope.lower() not in ["cid", "hostgroup"]:
+    log("The scope needs to be 'cid' or 'hostgroup'")
+    exit()
+
+
+#####################################
+## Main routine
+
+def main():
+    log("Starting execution of ProxyTool v3")
+
+    log("Authenticating to API")
+    auth = OAuth2(client_id=args.falcon_client_id, client_secret=args.falcon_client_secret, base_url=args.base_url)
+
+
+
+    ## Fetch list of hosts
+
+    if args.scope.lower() == "cid":
+        log("Getting all hosts from CID [" + args.scope_id + "]")
+        falcon = Hosts(auth_object=auth, base_url=args.base_url)
+    else:
+        log("Getting all hosts from host group ID [" + args.scope_id + "]")
+        falcon = HostGroup(auth_object=auth, base_url=args.base_url)
+
+
+    offset = ""
+    hosts_all = []
+
+    while True:
+        batch_size = 5000 ## 5000 is max supported by API
+        
+        if args.scope.lower() == "cid":
+            ## Fetch all Windows CID hosts
+            response = falcon.query_devices_by_filter_scroll(offset=offset, limit=batch_size, filter="platform_name:'Windows'")
+        else:
+            ## Fetch all Windows host group ID hosts
+            if offset == "":
+                response = falcon.query_group_members(limit=batch_size, filter="platform_name:'Windows'", id=args.scope_id)
+            else:
+                response = falcon.query_group_members(offset=offset, limit=batch_size, filter="platform_name:'Windows'", id=args.scope_id)
+
+        offset = response['body']['meta']['pagination']['offset']
+
+        for host_id in response['body']['resources']:
+            hosts_all.append(host_id)
+
+        log("-- Fetched " + str(len(response['body']['resources'])) + ' hosts, ' + str(len(hosts_all)) + '/' + str(response['body']['meta']['pagination']['total']))
+
+        if len(hosts_all) >= int(response['body']['meta']['pagination']['total']):
+            break
+
+    log("-- Retrieved a total of " + str(len(hosts_all)) + " hosts")
+
+
+    ## Now that we have the host IDs, we create a batch RTR list of commands to execute it in all hosts
+
+    falcon = RealTimeResponse(auth_object=auth, base_url=args.base_url)
+
+    ## Get batch id
+
+    response = falcon.batch_init_sessions(host_ids=hosts_all, queue_offline=True)
+    batch_id = response['body']['batch_id']
+
+    if batch_id:
+        log("Initiated RTR batch with id " + batch_id)
+    else:
+        exit()
+
+
+    ## Commands to change proxy config
+    ## Delete DisableProxy, PAC, PN, PP in both locations. Change CsProxyHostname and CsProxyport with new values
+
+
+    registry_stores = ["HKLM:\SYSTEM\Crowdstrike\{9b03c1d9-3138-44ed-9fae-d9f4c034b88d}\{16e0423f-7058-48c9-a204-725362b67639}\Default", 
+                        "HKLM:\SYSTEM\CurrentControlSet\Services\CSAgent\Sim"]
```

Comment:
These should be defined as raw strings due to the formatting character collisions.

```python
    registry_stores = [r"HKLM:\SYSTEM\Crowdstrike\{9b03c1d9-3138-44ed-9fae-d9f4c034b88d}\{16e0423f-7058-48c9-a204-725362b67639}\Default", 
                        r"HKLM:\SYSTEM\CurrentControlSet\Services\CSAgent\Sim"]
```